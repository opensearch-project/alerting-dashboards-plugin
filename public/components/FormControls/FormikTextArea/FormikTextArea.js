/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiTextArea } from '@elastic/eui';
import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikTextArea = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <TextArea name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <TextArea name={name} form={form} field={field} inputProps={inputProps} />
      )
    }
  />
);

const TextArea = ({ name, form, field, inputProps: { isInvalid, ...rest } }) => (
  <EuiTextArea
    isInvalid={typeof isInvalid === 'function' ? isInvalid(name, form) : isInvalid}
    {...field}
    {...rest}
  />
);

FormikTextArea.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

TextArea.propTypes = {
  name: PropTypes.string.isRequired,
  inputProps: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikTextArea;
