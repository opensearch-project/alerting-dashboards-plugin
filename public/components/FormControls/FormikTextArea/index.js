/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikTextArea from './FormikTextArea';

export default FormikTextArea;
