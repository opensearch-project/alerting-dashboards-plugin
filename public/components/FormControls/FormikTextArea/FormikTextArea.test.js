/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import FormikTextArea from './FormikTextArea';

describe('FormikTextArea', () => {
  test('renders', () => {
    const component = (
      <Formik>
        <FormikTextArea name="testing" />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
