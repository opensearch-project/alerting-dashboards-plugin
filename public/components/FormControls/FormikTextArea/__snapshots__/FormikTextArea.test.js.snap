// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikTextArea renders 1`] = `
<textarea
  class="euiTextArea euiTextArea--resizeVertical euiTextArea--compressed"
  name="testing"
  rows="3"
/>
`;
